Create variants of TestObservedConcurrency test

mgencur opened the pull request:
<!--
Request Prow to automatically lint any go code in this PR:

/lint
-->

Fixes https://github.com/knative/serving/issues/3518

## Proposed Changes

* Make it possible to run TestObservedConcurrency with different concurrency numbers

**Release Note**

<!-- Enter your extended release note in the below block. If the PR requires
additional action from users switching to the new release, include the string
"action required". If no release note is required, write "NONE". -->

```release-note

```

markusthoemmes commented on test/performance/observed_concurency_test.go:
```diff
@@ -121,6 +121,16 @@ func timeToScale(events []*event, start time.Time, desiredScale int) (time.Durat
 func TestObservedConcurrency(t *testing.T) {
+	tests := []int{5, 10} //going beyond 10 currently causes "overload" responses
+	for _, clients := range tests {
+		measureEachScale := clients <= 5 //for a few earliest scales we want to measure timeToScale for each step
```
Why do we need to distinguish this? Purely logging noise? I think for the performance tests it can be insightful to see when we reached which concurrency.

We also might not reach maximum concurrency at all.

mgencur replied:
Yeah. It is just that we don't print out every scale when we scale up to e.g. 50 or more. But I can remove this and measure every iteration regardless of the size.

mgencur commented:
OK. I've removed the option and all iterations will be reported.

mgencur commented:
/test pull-knative-serving-integration-tests

markusthoemmes commented on test/performance/observed_concurency_test.go:
```diff
@@ -136,7 +145,19 @@ func TestObservedConcurrency(t *testing.T) {
+				corev1.ResourceCPU:    resource.MustParse("10m"),
+				corev1.ResourceMemory: resource.MustParse("20Mi"),
+			},
+		},
```
I assume this is tuning down on the resource requirements of the app. Should that be done in the test or via config now that https://github.com/knative/serving/commit/306de3d6b7f2323e65bed38d6542c28e8bce642a has landed?

mgencur replied:
Yes. This is to cut down on resource requirements so as to allow for large-scale tests. I think the configuration should be done in the test, similar to all the other tests that manipulate the resource requirements (none of them use/change the new config map). We want to configure our own test and not default values affecting everyone else. Thanks for pointing out these new changes though.

markusthoemmes replied:
Would it suffice though to only specify the `Requests` section of it? Specifying such narrow limits seems a bit of an overkill to me.

mgencur replied:
OK. I've tested several times against a remote cluster and was reliably getting good responses but I guess it wouldn't hurt to get rid of the limit.

mgencur commented:
Hey, I've pushed one more commit that prevents errors when creating the XML file due to non-existent sub-directory. This new way creates just one resulting XML which is desired. Sorry for inconvenience.

srinivashegde86 commented on test/performance/observed_concurency_test.go:
```diff
@@ -121,6 +121,19 @@ func timeToScale(events []*event, start time.Time, desiredScale int) (time.Durat
+			tc = testConcurrencyN(t, clients, tc)
+		})
+	}
+	if err = testgrid.CreateXMLOutput(tc, t.Name()); err != nil {
```
```suggestion
	if err := testgrid.CreateXMLOutput(tc, t.Name()); err != nil {
```

mgencur replied:
Ah, missed that. Thanks

srinivashegde86 commented:
/test pull-knative-serving-upgrade-tests

srinivashegde86 commented:
/lgtm
/approve

markusthoemmes commented:
@mgencur please rebase. I'd think you can remove @srinivashegde86 commit, as you merely incooperated the review comment to resolve the CLA issue.

mgencur commented:
Rebased. I've also changed the testConcurrencyN method to create new slice of test cases that are returned instead of modifying the one passed as argument. Thanks for review!

mgencur commented:
One more little fix.

srinivashegde86 commented:
/lgtm
/approve
